src/rest_api.c

```diff
@@ -89,18 +166,34 @@ static int handler(struct mg_connection * conn, void * ignored)
 
 int auth_handler(struct mg_connection * conn, void * cbdata)
 {
-    int authorized = 1;
+    int authorized = 0;
+
+    point_t *public_key = (point_t *)cbdata;
+    if(public_key != NULL){
+        fprintf(stderr, "Public key is received: %s\n", public_key);
+    }
 
     char * auth_token = mg_get_header(conn, "Authorization");
     if(auth_token != NULL){
-        ///////////////////////////////////////////////////////
-        // TODO: Validate JWT TOKEN HERE
-        ///////////////////////////////////////////////////////
+        //Validate JWT TOKEN 
+        const char *token = auth_token + 7;  //skip prefix (Bearer)
+        fprintf(stderr, "Token received: %s\n", token);
+        //jwt_verify returns ecdsa_verify
+        //ecdsa_verify returns 1 when valid
+
+        if (jwt_verify(token, public_key) == 1) {
+            fprintf(stderr, "JWT token is verified!\n");
+            authorized = 1; 
+        } 
+    } 
+    else {
+        mg_send_http_error(conn, 401, "");
```

Comment:
It should not be here. If the  token is NULL, you send  `mg_send_http_error(conn, 401, "");`
but then also  `mg_send_http_error(conn, 403, "");`

Just don't send anything in this section.